shapely/strtree.py

```diff
@@ -75,26 +104,41 @@ class STRtree:
     None
     """
 
-    def __init__(self, geoms):
+    def __init__(self, initdata=None):
         warn(
             "STRtree will be completely changed in 2.0.0. The exact API is not yet decided, but will be documented before 1.8.0",
             ShapelyDeprecationWarning,
             stacklevel=2,
         )
-        # filter empty geometries out of the input
-        geoms = [geom for geom in geoms if not geom.is_empty]
-        self._n_geoms = len(geoms)
-
-        self._init_tree_handle(geoms)
-
-        # Keep references to geoms.
-        self._geoms = list(geoms)
-
-    def _init_tree_handle(self, geoms):
+        self._initdata = None
+        self._tree_handle = None
+        if initdata is not None:
+            self._initdata = []
+            for obj in initdata:
+                if not isinstance(obj, tuple):
+                    geom = obj
+                    value = obj
+                else:
+                    geom, value = obj
+                if not geom.is_empty:
+                    self._initdata.append(obj)
```

Comment:
Still saving a list of object references to keep things in the tree from getting GC'd.